CMakeLists.txt

```diff
@@ -113,6 +113,25 @@ if(CODECOVERAGE)
     message( "-- Building with llvm Code Coverage Tools")
 endif()
 
+if((NOT USE_CRYPTOPP_HASH) AND (NOT USE_OPENSSL_SHA_256) AND (NOT USE_OPENSSL_SHA3_256))
+    message(FATAL_ERROR "None of the cryptographic hashing libraries are enabled.")
+elseif(USE_OPENSSL_SHA_256 AND USE_OPENSSL_SHA3_256)
+    message(FATAL_ERROR "Cannot have OpenSSL's both SHA_256 and SHA3_256 hashing enabled.")
+elseif((USE_OPENSSL_SHA_256 AND USE_CRYPTOPP_HASH) OR (USE_OPENSSL_SHA3_256 AND USE_CRYPTOPP_HASH))
+    message(FATAL_ERROR "Cannot have both Cryptopp and OpenSSL hashing enabled.")
+endif()
+
+if(USE_OPENSSL_SHA_256)
+    message("-- USE_OPENSSL_SHA_256 Enabled")
+    string(APPEND CMAKE_CXX_FLAGS " -DUSE_OPENSSL_SHA_256")
+elseif(USE_OPENSSL_SHA3_256)
+    message("-- USE_OPENSSL_SHA3_256 Enabled")
+    string(APPEND CMAKE_CXX_FLAGS " -DUSE_OPENSSL_SHA3_256")
+elseif(USE_CRYPTOPP_HASH)
+    message("-- USE_CRYPTOPP_HASH Enabled")
+    string(APPEND CMAKE_CXX_FLAGS " -DUSE_CRYPTOPP_HASH")
+endif()
```

Comment:
Add a else for the case if none of these are defined and do FATAL_ERROR and remove the if((NOT USE_CRYPTOPP_HASH) AND (NOT USE_OPENSSL_SHA_256) AND (NOT USE_OPENSSL_SHA3_256)) in line no. 116

Reply:
Addressed https://github.com/arc-vmware/concord-bft/pull/2/commits/be7b4d3f850359b235887ae82cbe8f4c29a080c4